bundles/org.openhab.binding.tibber/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,112 @@
+		<description>Tibber API connector</description>
+		<channels>
+			<channel id="current_total" typeId="price" />
+			<channel id="current_startsAt" typeId="timestamp" />
+			<channel id="daily_from" typeId="timestamp" />
+			<channel id="daily_to" typeId="timestamp" />
+			<channel id="daily_cost" typeId="cost" />
+			<channel id="daily_consumption" typeId="consumption" />
+			<channel id="hourly_from" typeId="timestamp" />
+			<channel id="hourly_to" typeId="timestamp" />
+			<channel id="hourly_cost" typeId="cost" />
+			<channel id="hourly_consumption" typeId="consumption" />
+			<channel id="live_timestamp" typeId="timestamp" />
+			<channel id="live_power" typeId="power" />
+			<channel id="live_lastMeterConsumption" typeId="consumption" />
+			<channel id="live_accumulatedConsumption" typeId="consumption" />
+			<channel id="live_accumulatedCost" typeId="cost" />
+			<channel id="live_currency" typeId="currency" />
+			<channel id="live_minPower" typeId="power" />
+			<channel id="live_averagePower" typeId="power" />
+			<channel id="live_maxPower" typeId="power" />
+			<channel id="live_voltage1" typeId="voltage" />
+			<channel id="live_voltage2" typeId="voltage" />
+			<channel id="live_voltage3" typeId="voltage" />
+			<channel id="live_current1" typeId="current" />
+			<channel id="live_current2" typeId="current" />
+			<channel id="live_current3" typeId="current" />
+			<channel id="live_powerProduction" typeId="power" />
+			<channel id="live_minPowerproduction" typeId="power" />
+			<channel id="live_maxPowerproduction" typeId="power" />
+			<channel id="live_accumulatedProduction" typeId="production" />
+		</channels>
+		<properties>
+			<property name="vendor">Tibber</property>
+		</properties>
+		<config-description>
+			<parameter name="token" type="text" required="true"> 
+				<label>Token</label>
+				<description>Tibber Personal Token</description>
+			</parameter>
+			<parameter name="homeid" type="text" required="true"> 
+				<label>HomeID</label>
+				<description>Tibber Home ID</description>
+			</parameter>
+			<parameter name="refresh" type="integer" min="1" required="true"> 
+				<label>Refresh Interval</label>
+				<default>1</default>
+				<description>Manually Set Polling Frequency (minutes)</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+	<channel-type id="timestamp">
+		<item-type>DateTime</item-type>
+		<label>Timestamp</label>
+		<description>Timestamp for measurement/change</description>
+	</channel-type>
+	<channel-type id="price">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Current Total Price</label>
+		<description>Total Price: Energy + Tax</description>
+		<state pattern="%.3f %unit%"></state>
+	</channel-type>
+	<channel-type id="cost">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Total Cost</label>
+		<description>Cost at given time interval (e.g. hourly, daily, accumulated since midnight)</description>
+		<state pattern="%.3f %unit%"></state>
```

Comment:
You don't provide any units for this channel, and frankly openhab doesn't support any currency units right now so I'm not sure if there is any way you could reasonably resolve this right now...

Reply:
Uncertain what you mean? Seen several bindings use `Number:Dimensionless` with `state pattern = %unit%`

Reply:
There is nothing wrong with `Number:Dimensionless` and `state pattern = %unit%`. What is wrong is that any `Number:*` item-type requires the channel to use `QuantityType` and you are instead using `DecimalType` for this channel.

Reply:
I don't know if openhab will handle you using `DecimalType` instead of `QuantityType` but I do know that regardless it will not render properly in the UI. A bare minimum solution for now would be to remove `%unit%` from your pattern like so:
```suggestion
		<state pattern="%.3f"></state>
```

If you want a much better solution then you would have implement a `DynamicStateDescriptionProvider` to dynamically generate the pattern to be used for currency channels thus allowing utilize your currency unit channel to properly display the currency numbers.

Reply:
Thanks, DecimalType works fine (OH item state follows the exact same curve/trend on a time axis as API itself). Will chase the state pattern as a start.